src/main/webapp/app/code-editor/service/code-editor-repository.service.ts

```diff
@@ -91,27 +116,33 @@ export class CodeEditorRepositoryFileService extends DomainDependentEndpoint imp
     }
 
     getRepositoryContent = () => {
-        return this.http.get<{ [fileName: string]: FileType }>(`${this.restResourceUrl}/files`);
+        return this.http.get<{ [fileName: string]: FileType }>(`${this.restResourceUrl}/files`).pipe(this.handleErrorResponse<{ [fileName: string]: FileType }>());
     };
 
     getFile = (fileName: string) => {
-        return this.http.get(`${this.restResourceUrl}/file`, { params: new HttpParams().set('file', fileName), responseType: 'text' }).map(data => ({ fileContent: data }));
+        return this.http.get(`${this.restResourceUrl}/file`, { params: new HttpParams().set('file', fileName), responseType: 'text' }).pipe(
+            map(data => ({ fileContent: data })),
+            this.handleErrorResponse<{ fileContent: string }>(),
+        );
     };
 
     createFile = (fileName: string) => {
-        return this.http.post<void>(`${this.restResourceUrl}/file`, '', { params: new HttpParams().set('file', fileName) });
+        return this.http.post<void>(`${this.restResourceUrl}/file`, '', { params: new HttpParams().set('file', fileName) }).pipe(this.handleErrorResponse());
     };
 
     createFolder = (folderName: string) => {
-        return this.http.post<void>(`${this.restResourceUrl}/folder`, '', { params: new HttpParams().set('folder', folderName) });
+        return this.http.post<void>(`${this.restResourceUrl}/folder`, '', { params: new HttpParams().set('folder', folderName) }).pipe(this.handleErrorResponse());
     };
 
     updateFileContent = (fileName: string, fileContent: string) => {
-        return this.http.put(`${this.restResourceUrl}/file`, fileContent, {
-            params: new HttpParams().set('file', fileName),
-        });
+        return this.http
+            .put(`${this.restResourceUrl}/file`, fileContent, {
+                params: new HttpParams().set('file', fileName),
+            })
+            .pipe(this.handleErrorResponse());
     };
 
+    // TODO: We also have to handle receiving the conflict from the websocket
```

Comment:
Can you please resolve the TODO before we merge the PR

Reply:
This was also already resolved.